server/src/main/java/org/elasticsearch/search/fetch/subphase/ExplainPhase.java

```diff
@@ -40,6 +40,9 @@ public void process(HitContext hitContext) throws IOException {
                 for (RescoreContext rescore : context.rescore()) {
                     explanation = rescore.rescorer().explain(topLevelDocId, context.searcher(), rescore, explanation);
                 }
+                if (context.rankBuilder() != null) {
+                    explanation = context.rankBuilder().explainHit(explanation, hitContext.rankDoc(), context.queryNames());
```

Comment:
For every hit, `context.queryNames()` makes this call
```
searchContext.request().source().subSearches().stream().map(x -> x.getQueryBuilder().queryName()).toList();
```

This is unnecessarily expensive. Think you could statically set it in the processor and calculate it once?

Reply:
Updated!